core/inspector.rs

```diff
@@ -222,16 +221,12 @@ impl JsRuntimeInspector {
         // Do one "handshake" with a newly connected session at a time.
         if let Some(session) = &mut sessions.handshake {
           let poll_result = session.poll_unpin(cx);
-          let handshake_done =
-            replace(&mut self.flags.borrow_mut().session_handshake_done, false);
           match poll_result {
-            Poll::Pending if handshake_done => {
+            Poll::Pending => {
               let session = sessions.handshake.take().unwrap();
               sessions.established.push(session);
```

Comment:
This doesn't look correct to me. `session.poll_unpin(cx)` can return Pending multiple times.

Reply:
That's expected isn't it? Every future can return `Pending` multiple times. We're more interested when `Ready` is returned signaling that the session is done. As a side note, I would prefer to have explicit method like `pump_messages()` (which I actually refactored in https://github.com/denoland/deno/pull/12962) but that doesn't play nicely with `FuturesUnordered` which is used to store all sessions.